sdk/clientcore/core/src/main/java/io/clientcore/core/utils/UriBuilder.java

```diff
@@ -194,6 +194,27 @@ public UriBuilder addQueryParameter(String queryParameterName, String queryParam
         return this;
     }
 
+    /**
+     * Append the provided query parameter name and encoded values to query string for the final URI.
+     *
+     * @param queryParameterName The name of the query parameter.
+     * @param queryParameterEncodedValues The encoded values of the query parameter.
+     * @return The provided query parameter name and encoded values to query string for the final URI.
+     * @throws NullPointerException if {@code queryParameterName} or {@code queryParameterEncodedValues} are null.
+     */
+    public UriBuilder addQueryParameterValues(String queryParameterName, List<String> queryParameterEncodedValues) {
```

Comment:
I don't understand this. Are we using core util QueryParameter here? The annotation processor has a similar structure to other QueryParameters, so does it help with translation, or is this intentional?

Reply:
Yeah, this is adding to the `core` `QueryParameter` type, similar to what `addQueryParameter` on `UriBuilder` does but it adds a list of values in one call.